float32 roll
float32 pitch
uint8 i
uint8 j
float32[9] constraint
